code/game/machinery/computer/shuttle_computers.dm

```diff
@@ -296,15 +319,66 @@
 			to_chat(user, "<span class='warning'>No shuttle detected.</span>")
 		return
 
+	// Land on a planet
+	if(procgen_target && disk && istype(disk, /obj/item/weapon/disk/shuttle_coords/procedural))
+		var/obj/item/weapon/disk/shuttle_coords/procedural/proc_disk = disk
+		if(proc_disk.planet_ref)
+			travel_to_planet(proc_disk.planet_ref, user)
+			return
+
 	if(!selected_port && shuttle.docking_ports.len >= 2)
 		selected_port = pick(shuttle.docking_ports - shuttle.current_port)
 
+	// Close shuttle doors before departure
+	shuttle.close_all_doors()
```

Comment:
This could end up being the most noticeable change in your entire PR. For players who never use procgen planets, this will impact their workflow when using ships like the mining/research shuttles or station taxis.

This may sound silly but I think you should consider an implementation that determines if the doors *need* to be forced closed before the trip, OR to have a button on the shuttle console (default ON) that determines if the doors close every time the shuttle launches. This way, players will be less impacted and you won't get hatemail.

Note that the doors are closed *again* below in travel_to_planet.